### ISSUE TITLE
Workshop proposal: Hands-on workshop with AWS JavaScript SDK v3

### ISSUE THREAD

**Issue by trivikr** [seattlejs/seattlejs#179]
I'm creating this issue, as it was requested by several attendees during SeattleJS October meetup talk https://github.com/seattlejs/seattlejs/issues/176

## Title of workshop
Hands-on workshop with AWS JavaScript SDK v3

## Description
The AWS SDK for JavaScript enables developers to build libraries an applications that use AWS services. You can use the JavaScript API in the browser and inside Node.js applications on the server. The v3 is written in TypeScript, and adds several frequently requested features - like modular packages, better stack traces for debugging, middleware and more. In this workshop, you'll learn what's new in v3 by writing an example application.

## Learning Objectives
By the end of this workshop, you will:
* Learn how to update an existing v2 application to v3
* Use AWS services you're familiar with in v3
* Understand how modularization in v3 helps write cleaner code and improve performance by reducing bundle size
* Examine useful stack traces in v3 for faster debugging and reduce operational load.
* Inspect and customize request lifecycle using middleware in v3.

## Presenters:
* Trivikram Kamat
  * [Twitter](https://twitter.com/trivikram)
  * [Github](http://github.com/trivikr)
* Allan Zheng
  * [Twitter](https://twitter.com/allanzhengyp)
  * [Github](https://github.com/AllanFly120)

**Format**: 2-hour hands on workshop
**Target Audience**: Developers consuming AWS services with JavaScript/Node.js SDK
**Level**: Intermediate
**Prerequisites**: Basic experience of coding in JavaScript and/or Node.js

- [x] I've read and agree to the [SeattleJS Code of Conduct](https://github.com/seattlejs/seattlejs/blob/master/code-of-conduct.md)

**Comment by trivikr** [comment]
cc @AllanFly120 (co-presenter)

**Comment by bniedermeyer** [comment]
Hey @trivikr and @AllanFly120 We are still interested in doing this. Would sometime in January work? Would Amazon be willing to host?

**Comment by trivikr** [comment]
January would be perfect time to host this workshop!
I'll check if Amazon can host it.

**Comment by trivikr** [comment]
@bniedermeyer Is there a date you have in mind for the workshop?
Can we have this workshop in place of SeattleJS January meetup on Jan 9th ([link](https://www.meetup.com/seattlejs/events/clmbzqybccbmb/))?

**Comment by trivikr** [comment]
@bniedermeyer How many people can we expect for this workshop?

I'm trying for:
* Date: Thursday, February 20, 2020
* Time: 2pm-5pm (will check if 6pm-9pm option is available)
* Location: [Amazon Meeting Center (SEA45)](https://goo.gl/maps/1sFc3Hb8yNRwzfCf8)

Meeting Center sits 1000+ people, but I plan to request for classroom configuration and just one of the rooms - depending on how many people we can expect.

**Comment by bniedermeyer** [comment]
It's hard to say for sure since we don't do a ton of workshops. I'd say let's plan for 20 - 40 for now (or just one classroom's worth) and we can see how initial RSVPs look?

**Comment by trivikr** [comment]
> let's plan for 20 - 40 for now

Sounds good. If it's just 20-40, we don't have to plan in advance.
We can revisit this in mid or late-December.

**Comment by bniedermeyer** [comment]
Hi @trivikr - checking in on this. Are you still interested in doing the workshop sometime in Feb?

**Comment by trivikr** [comment]
Hi @bniedermeyer, we're interested in doing this workshop but Feb would be little early for us.
We'll respond by end-Feb on whether this workshop can be organized in March/April.

**Comment by bniedermeyer** [comment]
Sounds great, thanks!

**Comment by bniedermeyer** [comment]
Just checking in - do you think March or April will work best for you all?

**Comment by bniedermeyer** [comment]
Hey @trivikr. Do you have any interest in doing this remotely sometime in the next few months or would you rather wait until we are able to do in person?

**Comment by trivikr** [comment]
@bniedermeyer Closing this issue as workshop is not a priority for us right now.
The interested attendees can be directed to self-service repository https://github.com/aws-samples/aws-sdk-js-v3-workshop/

FYI: The instructions and code in the repository are going to be simplified further in coming weeks. WIP PR https://github.com/aws-samples/aws-sdk-js-v3-workshop/pull/176

**Comment by bniedermeyer** [comment]
Sounds great, thanks!
